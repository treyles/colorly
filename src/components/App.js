import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Upload from './Upload';
import Library from './Library';

const testData = [
  {
    url:
      'https://firebasestorage.googleapis.com/v0/b/colorly-44942.appspot.com/o/images%2Fa8140b0e-748a-4a8d-a9fe-91324f4174c5?alt=media&token=5ca4aef7-31cd-4389-b5f5-45da4292f057',
    palette: {
      color1: '#37BEB7'
    },
    title: 'Single Color',
    id: 'a8140b0e-748a-4a8d-a9fe-91324f4174c5'
  },
  {
    url:
      'https://firebasestorage.googleapis.com/v0/b/colorly-44942.appspot.com/o/images%2F2ec38480-3daf-4c9e-8a68-f2d9de46f6cb?alt=media&token=a8e97814-1650-48d7-9860-690fc936ecf5',
    palette: {
      color1: '#126794',
      color2: '#F8AC8E',
      color3: '#EF6868',
      color4: '#F2C61B',
      color5: '#DDDDDD'
    },
    title: 'Abstract Drawing',
    id: '2ec38480-3daf-4c9e-8a68-f2d9de46f6cb'
  },
  {
    url:
      'https://firebasestorage.googleapis.com/v0/b/colorly-44942.appspot.com/o/images%2F0917631b-26f3-46a3-804c-f5ecc1355e39?alt=media&token=d0e2052a-d3f8-4fbc-b430-b7ae259ed9f3',
    palette: {
      color1: '#49124B',
      color2: '#AE53A0',
      color3: '#F5966A',
      color4: '#F04267'
    },
    title: 'Uber Illustration Style',
    id: '0917631b-26f3-46a3-804c-f5ecc1355e39'
  }
];

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      library: testData
    };

    this.addPaletteToLibrary = this.addPaletteToLibrary.bind(this);
  }

  addPaletteToLibrary(obj) {
    const { library } = this.state;
    this.setState({
      library: [obj].concat(library)
    });
  }

  render() {
    const { library } = this.state;

    return (
      <BrowserRouter>
        <Switch>
          <Route
            exact
            path="/"
            render={() => <Library library={library} />}
          />
          <Route
            path="/upload"
            render={({ history }) => (
              <Upload
                history={history}
                addPaletteToLibrary={this.addPaletteToLibrary}
              />
            )}
          />
        </Switch>
      </BrowserRouter>
    );
  }
}

// <div>
//   <Header />
// <Upload addPaletteToLibrary={this.addPaletteToLibrary} />
// </div>
