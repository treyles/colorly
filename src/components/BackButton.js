import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../utils/Icon';

const BackButton = ({ onClick }) => (
  <button className="back" onClick={onClick}>
    <Icon icon="back" />
    <span>BACK TO PALETTES</span>
  </button>
);

BackButton.defaultProps = {
  onClick: () => {}
};

BackButton.propTypes = {
  onClick: PropTypes.func
};

export default BackButton;
