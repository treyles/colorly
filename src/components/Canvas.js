import React from 'react';
import { storage } from '../utils/base';

export default class Canvas extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      mousePosition: {},
      imageLoaded: false,
      previewColor: false,
      currentColor: null,
      winWidth: null,
      winHeight: null
    };

    this.handleColorAndPosition = this.handleColorAndPosition.bind(this);
    this.handleCanvasLeave = this.handleCanvasLeave.bind(this);
    this.handleColorPick = this.handleColorPick.bind(this);
    this.updateWindowDimensions = this.updateWindowDimensions.bind(this);
  }

  componentDidMount() {
    this.ctx = this.canvas.getContext('2d');
    this.updateWindowDimensions();
    window.addEventListener('resize', this.updateWindowDimensions);
    this.getImageFile();
  }

  updateWindowDimensions() {
    this.setState({
      winWidth: window.innerWidth,
      winHeight: window.innerHeight
    });
  }

  // TODO: name handleColorPreview ???
  handleColorAndPosition(e) {
    const { imageLoaded } = this.state;

    const rect = this.canvas.getBoundingClientRect();
    const actualX = Math.floor(e.clientX - rect.left);
    const actualY = Math.floor(e.clientY - rect.top);

    const pixelData = this.ctx.getImageData(actualX, actualY, 1, 1);
    const { data } = pixelData;
    const pixelColor = `rgba(${data[0]}, ${data[1]}, ${data[2]}, ${data[3]})`;

    if (imageLoaded) {
      this.setState({
        currentColor: pixelColor,
        mousePosition: { x: actualX, y: actualY },
        previewColor: true
      });
    }
  }

  handleCanvasLeave() {
    this.setState({
      previewColor: false
    });
  }

  handleColorPick() {
    const { currentColor } = this.state;
    this.props.makePalette(currentColor);
  }

  drawCanvasImage(image) {
    const heightPadding = 275;
    const widthPadding = 60;
    const containerWidth = this.state.winWidth - widthPadding;
    const containerHeight = this.state.winHeight - heightPadding;
    let imageWidth = image.width;
    let imageHeight = image.height;

    // if any part of the image (original dimensions) overflow
    // the container, shrink dimensions and maintain aspect ratio
    if (imageWidth > containerWidth || imageHeight > containerHeight) {
      // algorithm modified from 1owk3y: https://bit.ly/2JiYzHz
      const wFits = containerWidth / imageWidth;
      const hFits = containerHeight / imageHeight;
      const minFits = wFits > hFits ? hFits : wFits;

      imageWidth *= minFits;
      imageHeight *= minFits;
    }

    this.canvas.width = imageWidth;
    this.canvas.height = imageHeight;

    // TODO: check if this works
    this.ctx.webkitImageSmoothingEnabled = false;
    this.ctx.drawImage(image, 0, 0, imageWidth, imageHeight);
  }

  getImageFile() {
    const imageFile = storage
      .ref()
      .child(`images/${this.props.imageName}`);

    imageFile.getDownloadURL().then(url => {
      const image = new Image();
      image.crossOrigin = 'anonymous';

      image.onload = () => {
        this.drawCanvasImage(image);
        this.setState({ imageLoaded: true });
      };

      image.src = url;
    });
  }

  render() {
    const { mousePosition, previewColor, currentColor } = this.state;

    const containerStyle = {
      top: `${mousePosition.y}px`,
      left: `${mousePosition.x + 10}px`
    };

    const colorStyle = {
      background: currentColor
    };

    return (
      <div className="canvas-wrapper">
        <canvas
          ref={ref => {
            this.canvas = ref;
          }}
          // width={800}
          // height={550}
          onMouseMove={this.handleColorAndPosition}
          onMouseLeave={this.handleCanvasLeave}
          onClick={this.handleColorPick}
        />
        {previewColor && (
          <div className="preview-container" style={containerStyle}>
            <div className="preview-color" style={colorStyle} />
          </div>
        )}
      </div>
    );
  }
}
