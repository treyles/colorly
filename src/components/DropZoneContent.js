import React from 'react';
import PropTypes from 'prop-types';

export default class DropZoneContent extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange() {
    this.props.handleImageDrop();
  }

  render() {
    const { imageAlert } = this.props;

    return (
      <div className="drop-zone-content">
        <label htmlFor="file-browse" className={imageAlert && 'alert'}>
          {imageAlert ? (
            <h1>{imageAlert}</h1>
          ) : (
            <div>
              <h1>Drag and Drop Image Here</h1>
              <h3>JPG, GIF, PNG or SVG no larger than 3 MB</h3>
            </div>
          )}
        </label>
        <input
          id="file-browse"
          type="file"
          accept="image/*"
          onChange={this.handleChange}
        />
      </div>
    );
  }
}

DropZoneContent.defaultProps = {
  imageAlert: null
};

DropZoneContent.propTypes = {
  imageAlert: PropTypes.string,
  handleImageDrop: PropTypes.func.isRequired
};
