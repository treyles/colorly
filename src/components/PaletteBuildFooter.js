import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import Icon from '../utils/Icon';
import {
  setCheckedColor,
  setPaletteTitle,
  clearPalette,
  addCardToLibrary
} from '../actions';

class PaletteBuildFooter extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      submitAlert: false
    };

    this.handleTitleIconClick = this.handleTitleIconClick.bind(this);
    this.handleKeyDown = this.handleKeyDown.bind(this);
    this.setTitleInputRef = this.setTitleInputRef.bind(this);
    this.handleSaveClick = this.handleSaveClick.bind(this);
  }

  componentWillUnmount() {
    clearTimeout(this.submitAlertTimer);
  }

  setTitleInputRef(ref) {
    this.titleInput = ref;
  }

  handleTitleIconClick() {
    this.titleInput.focus();
  }

  handleKeyDown(e) {
    if (e.keyCode === 13) {
      this.titleInput.blur();
    }
  }

  handleSaveClick() {
    const { palette, title, history } = this.props;

    if (!Object.keys(palette).length || !title.length) {
      this.activateSubmitAlert();
      return;
    }

    this.props.addCardToLibrary(history);
  }

  activateSubmitAlert() {
    const { palette } = this.props;
    let { submitAlert } = this.state;

    if (!Object.keys(palette).length) {
      submitAlert = 'Palette Is Empty!';
    } else {
      submitAlert = 'Title Required!';
    }

    this.setState({ submitAlert });
    this.resetSubmitAlert();
  }

  resetSubmitAlert() {
    this.submitAlertTimer = setTimeout(() => {
      this.setState({ submitAlert: false });
    }, 1500);
  }

  // TODO: save for last (if have time), loop palette color markup
  render() {
    const { submitAlert } = this.state;
    const {
      palette,
      checked,
      title,
      library,
      clearPalette,
      setCheckedColor,
      setPaletteTitle
    } = this.props;

    return (
      <div className="footer">
        <div className="palette-name">
          <span
            onClick={this.handleTitleIconClick}
            onKeyDown={this.handleTitleIconClick}
            role="button"
            tabIndex="0"
          >
            <Icon icon="pencil" />
          </span>
          <input
            ref={this.setTitleInputRef}
            onChange={e => setPaletteTitle(e)}
            value={title}
            className="palette-input"
            type="text"
            placeholder={`Color Palette #${library.length + 1}`}
            maxLength="24"
            onKeyDown={this.handleKeyDown}
          />
        </div>
        <div className="palette-colors">
          <input
            type="radio"
            id="color1"
            checked={checked === 'color1'}
            onChange={e => setCheckedColor(e)}
          />
          <label
            className="radio"
            htmlFor="color1"
            style={{ background: palette.color1 }}
          />
          <input
            type="radio"
            id="color2"
            checked={checked === 'color2'}
            onChange={e => setCheckedColor(e)}
          />
          <label
            className="radio"
            htmlFor="color2"
            style={{ background: palette.color2 }}
          />
          <input
            type="radio"
            id="color3"
            checked={checked === 'color3'}
            onChange={e => setCheckedColor(e)}
          />
          <label
            className="radio"
            htmlFor="color3"
            style={{ background: palette.color3 }}
          />
          <input
            type="radio"
            id="color4"
            checked={checked === 'color4'}
            onChange={e => setCheckedColor(e)}
          />
          <label
            className="radio"
            htmlFor="color4"
            style={{ background: palette.color4 }}
          />
          <input
            type="radio"
            id="color5"
            checked={checked === 'color5'}
            onChange={e => setCheckedColor(e)}
          />
          <label
            className="radio"
            htmlFor="color5"
            style={{ background: palette.color5 }}
          />
        </div>
        <div className="palette-buttons">
          <button className="clear" onClick={() => clearPalette()}>
            RESET
          </button>
          <button className="save" onClick={this.handleSaveClick}>
            SAVE PALETTE
            <div className={`submit-alert ${submitAlert ? 'active' : ''}`}>
              {submitAlert}
            </div>
          </button>
        </div>
      </div>
    );
  }
}

PaletteBuildFooter.propTypes = {
  history: PropTypes.object.isRequired, // eslint-disable-line
  // action creators
  setCheckedColor: PropTypes.func.isRequired,
  setPaletteTitle: PropTypes.func.isRequired,
  clearPalette: PropTypes.func.isRequired,
  addCardToLibrary: PropTypes.func.isRequired,
  // from store
  library: PropTypes.arrayOf(PropTypes.object).isRequired,
  checked: PropTypes.string.isRequired,
  palette: PropTypes.objectOf(PropTypes.string).isRequired,
  title: PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  library: state.data.library,
  checked: state.build.checked,
  palette: state.build.card.palette,
  title: state.build.card.title
});

export default withRouter(
  connect(mapStateToProps, {
    setCheckedColor,
    setPaletteTitle,
    clearPalette,
    addCardToLibrary
  })(PaletteBuildFooter)
);
