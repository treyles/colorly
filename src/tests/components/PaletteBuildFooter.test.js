import React from 'react';
import { shallow, mount } from 'enzyme';
import { PaletteBuildFooter } from '../../components/PaletteBuildFooter';

describe('PaletteBuildFooter', () => {
  const props = {
    setCheckedColor: jest.fn(),
    setPaletteTitle: jest.fn(),
    addCardToLibrary: jest.fn(),
    clearPalette: jest.fn(),
    palette: {},
    checked: 'color1',
    title: '',
    library: [],
    history: {}
  };

  it('renders correctly', () => {
    const component = shallow(<PaletteBuildFooter {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('calls setCheckedColor on palette selection', () => {
    const component = shallow(<PaletteBuildFooter {...props} />);

    component
      .find('input')
      .at(1)
      .simulate('change');
    expect(props.setCheckedColor).toHaveBeenCalled();
  });

  it('calls setPaletteTitle when entering title in text field', () => {
    const component = shallow(<PaletteBuildFooter {...props} />);

    component.find('.palette-input').simulate('change');
    expect(props.setPaletteTitle).toHaveBeenCalled();
  });

  it('focuses on text field when icon is clicked', () => {
    const component = mount(<PaletteBuildFooter {...props} />);
    const element = component.instance().titleInput;
    jest.spyOn(element, 'focus');

    component
      .find('span')
      .at(0)
      .simulate('click');

    expect(element.focus).toHaveBeenCalled();
  });

  it('calls handleSaveClick when save button is clicked', () => {
    const spy = jest
      .spyOn(PaletteBuildFooter.prototype, 'handleSaveClick')
      .mockImplementation(jest.fn());
    const component = shallow(<PaletteBuildFooter {...props} />);
    component.find('.save').simulate('click');
    expect(spy).toHaveBeenCalled();
  });

  it('calls clearPalette() when clear button is clicked', () => {
    const component = shallow(<PaletteBuildFooter {...props} />);
    component.find('.clear').simulate('click');
    expect(props.clearPalette).toHaveBeenCalled();
  });
});
