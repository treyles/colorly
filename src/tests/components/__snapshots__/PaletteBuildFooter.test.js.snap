// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteBuildFooter renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PaletteBuildFooter
    addCardToLibrary={[MockFunction]}
    checked="color1"
    clearPalette={[MockFunction]}
    history={Object {}}
    library={Array []}
    palette={Object {}}
    setCheckedColor={[MockFunction]}
    setPaletteTitle={[MockFunction]}
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="palette-name"
        >
          <span
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <Icon
              icon="pencil"
            />
          </span>
          <input
            className="palette-input"
            maxLength="24"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Color Palette #1"
            type="text"
            value=""
          />
        </div>,
        <div
          className="palette-colors"
        >
          <input
            checked={true}
            id="color1"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="radio"
            htmlFor="color1"
            style={
              Object {
                "background": undefined,
              }
            }
          />
          <input
            checked={false}
            id="color2"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="radio"
            htmlFor="color2"
            style={
              Object {
                "background": undefined,
              }
            }
          />
          <input
            checked={false}
            id="color3"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="radio"
            htmlFor="color3"
            style={
              Object {
                "background": undefined,
              }
            }
          />
          <input
            checked={false}
            id="color4"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="radio"
            htmlFor="color4"
            style={
              Object {
                "background": undefined,
              }
            }
          />
          <input
            checked={false}
            id="color5"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="radio"
            htmlFor="color5"
            style={
              Object {
                "background": undefined,
              }
            }
          />
        </div>,
        <div
          className="palette-buttons"
        >
          <button
            className="clear"
            onClick={[Function]}
          >
            RESET
          </button>
          <button
            className="save"
            onClick={[Function]}
          >
            SAVE PALETTE
            <div
              className="submit-alert "
            />
          </button>
        </div>,
      ],
      "className": "footer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="0"
            >
              <Icon
                icon="pencil"
              />
            </span>,
            <input
              className="palette-input"
              maxLength="24"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Color Palette #1"
              type="text"
              value=""
            />,
          ],
          "className": "palette-name",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                icon="pencil"
              />,
              "onClick": [Function],
              "onKeyDown": [Function],
              "role": "button",
              "tabIndex": "0",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "icon": "pencil",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "palette-input",
              "maxLength": "24",
              "onChange": [Function],
              "onKeyDown": [Function],
              "placeholder": "Color Palette #1",
              "type": "text",
              "value": "",
            },
            "ref": [Function],
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={true}
              id="color1"
              onChange={[Function]}
              type="radio"
            />,
            <label
              className="radio"
              htmlFor="color1"
              style={
                Object {
                  "background": undefined,
                }
              }
            />,
            <input
              checked={false}
              id="color2"
              onChange={[Function]}
              type="radio"
            />,
            <label
              className="radio"
              htmlFor="color2"
              style={
                Object {
                  "background": undefined,
                }
              }
            />,
            <input
              checked={false}
              id="color3"
              onChange={[Function]}
              type="radio"
            />,
            <label
              className="radio"
              htmlFor="color3"
              style={
                Object {
                  "background": undefined,
                }
              }
            />,
            <input
              checked={false}
              id="color4"
              onChange={[Function]}
              type="radio"
            />,
            <label
              className="radio"
              htmlFor="color4"
              style={
                Object {
                  "background": undefined,
                }
              }
            />,
            <input
              checked={false}
              id="color5"
              onChange={[Function]}
              type="radio"
            />,
            <label
              className="radio"
              htmlFor="color5"
              style={
                Object {
                  "background": undefined,
                }
              }
            />,
          ],
          "className": "palette-colors",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": true,
              "id": "color1",
              "onChange": [Function],
              "type": "radio",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "radio",
              "htmlFor": "color1",
              "style": Object {
                "background": undefined,
              },
            },
            "ref": null,
            "rendered": null,
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "id": "color2",
              "onChange": [Function],
              "type": "radio",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "radio",
              "htmlFor": "color2",
              "style": Object {
                "background": undefined,
              },
            },
            "ref": null,
            "rendered": null,
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "id": "color3",
              "onChange": [Function],
              "type": "radio",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "radio",
              "htmlFor": "color3",
              "style": Object {
                "background": undefined,
              },
            },
            "ref": null,
            "rendered": null,
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "id": "color4",
              "onChange": [Function],
              "type": "radio",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "radio",
              "htmlFor": "color4",
              "style": Object {
                "background": undefined,
              },
            },
            "ref": null,
            "rendered": null,
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "id": "color5",
              "onChange": [Function],
              "type": "radio",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "radio",
              "htmlFor": "color5",
              "style": Object {
                "background": undefined,
              },
            },
            "ref": null,
            "rendered": null,
            "type": "label",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="clear"
              onClick={[Function]}
            >
              RESET
            </button>,
            <button
              className="save"
              onClick={[Function]}
            >
              SAVE PALETTE
              <div
                className="submit-alert "
              />
            </button>,
          ],
          "className": "palette-buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "RESET",
              "className": "clear",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "RESET",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "SAVE PALETTE",
                <div
                  className="submit-alert "
                />,
              ],
              "className": "save",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "SAVE PALETTE",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": false,
                  "className": "submit-alert ",
                },
                "ref": null,
                "rendered": false,
                "type": "div",
              },
            ],
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="palette-name"
          >
            <span
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="0"
            >
              <Icon
                icon="pencil"
              />
            </span>
            <input
              className="palette-input"
              maxLength="24"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Color Palette #1"
              type="text"
              value=""
            />
          </div>,
          <div
            className="palette-colors"
          >
            <input
              checked={true}
              id="color1"
              onChange={[Function]}
              type="radio"
            />
            <label
              className="radio"
              htmlFor="color1"
              style={
                Object {
                  "background": undefined,
                }
              }
            />
            <input
              checked={false}
              id="color2"
              onChange={[Function]}
              type="radio"
            />
            <label
              className="radio"
              htmlFor="color2"
              style={
                Object {
                  "background": undefined,
                }
              }
            />
            <input
              checked={false}
              id="color3"
              onChange={[Function]}
              type="radio"
            />
            <label
              className="radio"
              htmlFor="color3"
              style={
                Object {
                  "background": undefined,
                }
              }
            />
            <input
              checked={false}
              id="color4"
              onChange={[Function]}
              type="radio"
            />
            <label
              className="radio"
              htmlFor="color4"
              style={
                Object {
                  "background": undefined,
                }
              }
            />
            <input
              checked={false}
              id="color5"
              onChange={[Function]}
              type="radio"
            />
            <label
              className="radio"
              htmlFor="color5"
              style={
                Object {
                  "background": undefined,
                }
              }
            />
          </div>,
          <div
            className="palette-buttons"
          >
            <button
              className="clear"
              onClick={[Function]}
            >
              RESET
            </button>
            <button
              className="save"
              onClick={[Function]}
            >
              SAVE PALETTE
              <div
                className="submit-alert "
              />
            </button>
          </div>,
        ],
        "className": "footer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex="0"
              >
                <Icon
                  icon="pencil"
                />
              </span>,
              <input
                className="palette-input"
                maxLength="24"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Color Palette #1"
                type="text"
                value=""
              />,
            ],
            "className": "palette-name",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  icon="pencil"
                />,
                "onClick": [Function],
                "onKeyDown": [Function],
                "role": "button",
                "tabIndex": "0",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "icon": "pencil",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "palette-input",
                "maxLength": "24",
                "onChange": [Function],
                "onKeyDown": [Function],
                "placeholder": "Color Palette #1",
                "type": "text",
                "value": "",
              },
              "ref": [Function],
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={true}
                id="color1"
                onChange={[Function]}
                type="radio"
              />,
              <label
                className="radio"
                htmlFor="color1"
                style={
                  Object {
                    "background": undefined,
                  }
                }
              />,
              <input
                checked={false}
                id="color2"
                onChange={[Function]}
                type="radio"
              />,
              <label
                className="radio"
                htmlFor="color2"
                style={
                  Object {
                    "background": undefined,
                  }
                }
              />,
              <input
                checked={false}
                id="color3"
                onChange={[Function]}
                type="radio"
              />,
              <label
                className="radio"
                htmlFor="color3"
                style={
                  Object {
                    "background": undefined,
                  }
                }
              />,
              <input
                checked={false}
                id="color4"
                onChange={[Function]}
                type="radio"
              />,
              <label
                className="radio"
                htmlFor="color4"
                style={
                  Object {
                    "background": undefined,
                  }
                }
              />,
              <input
                checked={false}
                id="color5"
                onChange={[Function]}
                type="radio"
              />,
              <label
                className="radio"
                htmlFor="color5"
                style={
                  Object {
                    "background": undefined,
                  }
                }
              />,
            ],
            "className": "palette-colors",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": true,
                "id": "color1",
                "onChange": [Function],
                "type": "radio",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "radio",
                "htmlFor": "color1",
                "style": Object {
                  "background": undefined,
                },
              },
              "ref": null,
              "rendered": null,
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "id": "color2",
                "onChange": [Function],
                "type": "radio",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "radio",
                "htmlFor": "color2",
                "style": Object {
                  "background": undefined,
                },
              },
              "ref": null,
              "rendered": null,
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "id": "color3",
                "onChange": [Function],
                "type": "radio",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "radio",
                "htmlFor": "color3",
                "style": Object {
                  "background": undefined,
                },
              },
              "ref": null,
              "rendered": null,
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "id": "color4",
                "onChange": [Function],
                "type": "radio",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "radio",
                "htmlFor": "color4",
                "style": Object {
                  "background": undefined,
                },
              },
              "ref": null,
              "rendered": null,
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "id": "color5",
                "onChange": [Function],
                "type": "radio",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "radio",
                "htmlFor": "color5",
                "style": Object {
                  "background": undefined,
                },
              },
              "ref": null,
              "rendered": null,
              "type": "label",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="clear"
                onClick={[Function]}
              >
                RESET
              </button>,
              <button
                className="save"
                onClick={[Function]}
              >
                SAVE PALETTE
                <div
                  className="submit-alert "
                />
              </button>,
            ],
            "className": "palette-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "RESET",
                "className": "clear",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "RESET",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "SAVE PALETTE",
                  <div
                    className="submit-alert "
                  />,
                ],
                "className": "save",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "SAVE PALETTE",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": false,
                    "className": "submit-alert ",
                  },
                  "ref": null,
                  "rendered": false,
                  "type": "div",
                },
              ],
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
